library/core/src/mem/maybe_uninit.rs

```diff
@@ -910,52 +910,6 @@ impl<T> MaybeUninit<T> {
         }
     }
 
-    /// Extracts the values from an array of `MaybeUninit` containers.
-    ///
-    /// # Safety
-    ///
-    /// It is up to the caller to guarantee that all elements of the array are
-    /// in an initialized state.
-    ///
-    /// # Examples
-    ///
-    /// ```
-    /// #![feature(maybe_uninit_uninit_array)]
-    /// #![feature(maybe_uninit_array_assume_init)]
-    /// use std::mem::MaybeUninit;
-    ///
-    /// let mut array: [MaybeUninit<i32>; 3] = MaybeUninit::uninit_array();
-    /// array[0].write(0);
-    /// array[1].write(1);
-    /// array[2].write(2);
-    ///
-    /// // SAFETY: Now safe as we initialised all elements
-    /// let array = unsafe {
-    ///     MaybeUninit::array_assume_init(array)
-    /// };
-    ///
-    /// assert_eq!(array, [0, 1, 2]);
-    /// ```
-    #[unstable(feature = "maybe_uninit_array_assume_init", issue = "96097")]
-    #[rustc_const_unstable(feature = "const_maybe_uninit_array_assume_init", issue = "96097")]
-    #[inline(always)]
-    #[track_caller]
-    pub const unsafe fn array_assume_init<const N: usize>(array: [Self; N]) -> [T; N] {
-        // SAFETY:
-        // * The caller guarantees that all elements of the array are initialized
-        // * `MaybeUninit<T>` and T are guaranteed to have the same layout
-        // * `MaybeUninit` does not drop, so there are no double-frees
-        // And thus the conversion is safe
-        let ret = unsafe {
-            intrinsics::assert_inhabited::<[T; N]>();
```

Comment:
I see that `transpose` does not have this assertion. Could you add it there as well?

Reply:
Nope! :) The whole point of transpose is that it's completely safe because it only converts `MaybeUninit<[T; N]>` <-> `[MaybeUninit<T>; N]`. The `assert_inhabited` will come back when the user does `assume_init` after the transpose.

Reply:
Oh, I forgot about that. `assume_init` does have the assertion, so all good, my bad.